ERCS/erc-7855.md

```diff
@@ -0,0 +1,149 @@
+author: Amitesh Gautam (@blockchainDevAmitesh) <blockchaindev.amitesh@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7849-fungible-token-with-gas-rebates/22356
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-01-01
+---
+
+## Abstract
+
+This proposal introduces a new fungible token standard focused on security, efficiency, and interoperability. It extends traditional [ERC-20](./eip-20.md) functionality with built-in gas rebates, secure signature validation using [EIP-712](./eip-712.md), and optimized transfer mechanisms. The standard includes protections against common attack vectors while maintaining full compatibility with existing token interfaces.
+
+## Motivation
+
+Current fungible token standards lack built-in security features and gas optimization mechanisms. This proposal addresses these limitations by incorporating:
+- Native gas rebate system to reduce transaction costs
+- [EIP-712](./eip-712.md) compliant signatures for gasless approvals
+- Strong security measures against common vulnerabilities
+- Efficient state management and error handling
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Core Variables
+- `name`: Token name
+- `symbol`: Token symbol
+- `decimals`: Number of decimal places
+- `totalSupply`: Total token supply
+- `_balances`: Mapping of address to token balance
+- `_allowances`: Mapping of owner address to spender address to allowed amount
+
+### Features
+- Gas rebate system
+- [EIP-712](./eip-712.md) signature validation
+- Nonce tracking for replay protection
+
+### Functions
+
+#### View Functions
+
+**`name()`**
+- Returns the token name
+- Type: `string`
+
+**`symbol()`**
+- Returns the token symbol
+- Type: `string`
+
+**`decimals()`**
+- Returns number of decimals
+- Type: `uint8`
+
+**`totalSupply()`**
+- Returns total token supply
+- Type: `uint256`
+
+**`balanceOf(address account)`**
+- Returns token balance of specified account
+- Type: `uint256`
+
+**`allowance(address owner, address spender)`**
+- Returns remaining allowance for spender
+- Type: `uint256`
+
+#### State-Changing Functions
+
+**`transfer(address to, uint256 amount)`**
+- Transfers tokens to specified address
+- Includes gas rebate processing
+- Returns: `bool`
+
+**`approve(address spender, uint256 amount)`**
+- Approves spender to spend tokens
+- Returns: `bool`
+
+**`transferFrom(address from, address to, uint256 amount)`**
+- Transfers tokens between addresses
+- Requires sufficient allowance
+- Returns: `bool`
+
+**`permit(address owner, address spender, uint256 value, uint256 deadline, uint8 v, bytes32 r, bytes32 s)`**
+- Processes gasless approvals using [EIP-712](./eip-712.md) signatures
+- Validates signature and deadline
+- No return value
```

Comment:
You need to define a lot more here than just "using EIP-712 signatures". What's the domain separator? How are the parameters encoded?